ten-simple-rules-dockerfiles.Rmd

```diff
@@ -34,57 +35,57 @@ csl: plos.csl
-Containerisation can help with these needs.
-It provides portable computational environments that are built from human and machine readable instructions and are thus well documented and usable by many.
-These environments can then _host_ executable and human-readable notebooks with the scientific analyses [@rule_ten_2019] and make sharing of properly reproducible workflows (cf. Rule 10 of @sandve_ten_2013) much easier as the environments can closely resemble the original researchers virtual laboratory.
-
-While there are a few tutorials for using containers for reproducible research [^1], there is no extensive examination how to write the actual instructions to create the containers.
-Several platforms for facilitating reproducible research are built on top of containers [@brinckman_computing_2018; @code_ocean_2019; @simko_reana_2019; @jupyter_binder_2018; @nust_opening_2017], but usually and intentionally hide this complexity from the researcher.
-Not everybody needs to understand and write Dockerfiles, as is reflected in rule 1.
-However, as one can argue that _"the number of unique reserach environments approximates the number of researchers"_ [@nust_opening_2017], the researchers who do need to craft their own environment definition should follow practices taking into account the scientific context.
+With access to version control systems [REF GITHUB, GITLAB, etc.] it has become increasingly easy to not only share algorithms, but also instructions for building and testing.
+Within this content, it is suggested to include instructions for building containers to package software and data dependencies [@boettiger_introduction_2017].
+By way of providing these instructions along with thorough documentation, it's much more likely to be able to reproduce an analysis workflow.
+A container is a portabal, encapsulated environment that is built from human and machine readable instructions.
+For example, containers have been created to host scientific notebooks [@rule_ten_2019], and share reproducible workflows (cf. Rule 10 of @sandve_ten_2013).
+<!-- the virtual laboratory reference isn't great, I removed it-->
+
+While there are several tutorials for using containers for reproducible research [^1], there is no extensive examination for how to write the actual instructions to create the containers.
+Several platforms for facilitating reproducible research are built on top of containers [@brinckman_computing_2018; @code_ocean_2019; @simko_reana_2019; @jupyter_binder_2018; @nust_opening_2017], but they hide any complexity from the researcher.
+While not everybody needs to understand and write Dockerfiles, as _"the number of unique research environments approximates the number of researchers"_ [@nust_opening_2017], the researchers who do need to craft their own environment definition should follow good practices.
```

Comment:
There is also scope for folk being able to "read and modify" pre-existing Dockerfiles. Good practice in writing eg using conventions, makes things easier to read; pragmatically, for folk wishing to make changes, sticking to conventions can also help with the writing/modification even if they aren't sure how things actually work...

Reply:
I agree with this point and will take a look again if it is mentioned clearly enough.

Reply:
Tried to clarify the advantage of conventions in this change to the introduction: https://github.com/nuest/ten-simple-rules-dockerfiles/commit/b83bc16a08d90d609c4d74c1ac59b5a58eba6077